diff --git a/generators/fstab.go b/generators/fstab.go
index bb77e4b..c581415 100644
--- a/generators/fstab.go
+++ b/generators/fstab.go
@@ -8,6 +8,7 @@ import (
 
 	"github.com/lxc/distrobuilder/v3/image"
 	"github.com/lxc/distrobuilder/v3/shared"
+	incusArch "github.com/lxc/incus/v7/shared/osarch"
 )
 
 type fstab struct {
@@ -28,10 +29,6 @@ func (g *fstab) RunIncus(img *image.IncusImage, target shared.DefinitionTargetIn
 
 	defer f.Close()
 
-	content := `LABEL=rootfs  /         %s  %s  0 0
-LABEL=UEFI    /boot/efi vfat  defaults  0 0
-`
-
 	fs := target.VM.Filesystem
 
 	if fs == "" {
@@ -44,7 +41,20 @@ LABEL=UEFI    /boot/efi vfat  defaults  0 0
 		options = fmt.Sprintf("%s,subvol=@", options)
 	}
 
-	_, err = fmt.Fprintf(f, content, fs, options)
+	content := fmt.Sprintf(
+		"LABEL=rootfs  /         %s  %s  0 0\n",
+		fs, options,
+	)
+
+	archID, err := incusArch.ArchitectureID(img.Architecture())
+	if err != nil {
+		return err
+	}
+	if archID != incusArch.ARCH_64BIT_POWERPC_LITTLE_ENDIAN {
+		content += "LABEL=UEFI    /boot/efi vfat  defaults  0 0\n"
+	}
+
+	_, err = f.WriteString(content)
 	if err != nil {
 		return fmt.Errorf("Failed to write string to file %q: %w", filepath.Join(g.sourceDir, "etc/fstab"), err)
 	}
diff --git a/image/incus.go b/image/incus.go
index cc60cea..c007972 100644
--- a/image/incus.go
+++ b/image/incus.go
@@ -205,3 +205,8 @@ func (l *IncusImage) createMetadata() error {
 
 	return nil
 }
+
+// Architecture returns the image architecture.
+func (l *IncusImage) Architecture() string {
+	return l.definition.Image.Architecture
+}
